src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3122,13 +3122,14 @@ void addCloneMethod()
 
             ImmutableArray<PropertySymbol> addProperties(ImmutableArray<ParameterSymbol> recordParameters)
             {
+                Debug.Assert(recordParameters.Length == paramList.Parameters.Count);
                 var existingOrAddedMembers = ArrayBuilder<PropertySymbol>.GetInstance(recordParameters.Length);
                 int addedCount = 0;
-                foreach (ParameterSymbol param in recordParameters)
+                for (int i = 0; i < recordParameters.Length; i++)
                 {
+                    var param = recordParameters[i];
                     bool isInherited = false;
-                    var syntax = param.GetNonNullSyntaxNode();
```

Comment:
>GetNonNullSyntaxNode [](start = 39, length = 20)

I think we should make the ```GetNonNullSyntaxNode``` work properly. It could be used elsewhere and, by looking at the implementation, I would expect it to work.